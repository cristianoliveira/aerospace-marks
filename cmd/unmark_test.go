package cmd

import (
	"strings"
	"testing"

	"github.com/cristianoliveira/aerospace-marks/internal/mocks"
	"github.com/cristianoliveira/aerospace-marks/internal/storage"
	"github.com/cristianoliveira/aerospace-marks/internal/testutils"
	"github.com/gkampitakis/go-snaps/snaps"
	"go.uber.org/mock/gomock"
)

func TestUnmarkCommand(t *testing.T) {
	t.Run("unmarks a mark from a window - `marks unmark mark1`", func(t *testing.T) {
		// t.Skip("Skipping")
		command := "unmark"
		args := []string{command, "mark1"}

		ctrl := gomock.NewController(t)
		defer ctrl.Finish()

		storageDbClient, strg := mocks.MockStorageDbClient(ctrl)
		dbResult := mocks.MockStorageDbResult(ctrl, nil, &[]int64{1}[0])
		gomock.InOrder(
			storageDbClient.EXPECT().
				Execute(strings.TrimSpace(`
					DELETE FROM marks WHERE mark = ?
				`),
				"mark1").
				Return(dbResult, nil).
				Times(1),
		)

		out, err := testutils.CmdExecute(NewRootCmd(strg), args...)
		if err != nil {
			t.Fatal(err)
		}

		cmdAsString := "aerospace-marks " + strings.Join(args, " ") + "\n"
		snaps.MatchSnapshot(t, cmdAsString, out)
	})

	t.Run("unmarks all marks from a window - `marks unmark`", func(t *testing.T) {
		// t.Skip("Skipping")
		command := "unmark"
		args := []string{command}

		ctrl := gomock.NewController(t)
		defer ctrl.Finish()

		storageDbClient, strg := mocks.MockStorageDbClient(ctrl)
		dbResult := mocks.MockStorageDbResult(ctrl, nil, &[]int64{2}[0])
		gomock.InOrder(
			storageDbClient.EXPECT().
				Execute(`DELETE FROM marks`).
				Return(dbResult, nil).
				Times(1),
		)

		out, err := testutils.CmdExecute(NewRootCmd(strg), args...)
		if err != nil {
			t.Fatal(err)
		}

		cmdAsString := "aerospace-marks " + strings.Join(args, " ") + "\n"
		snaps.MatchSnapshot(t, cmdAsString, out)
	})

	t.Run("unmarks --help", func(t *testing.T) {
		// t.Skip("Skipping")
		command := "unmark"
		args := []string{command, "--help"}
		connector := storage.MarksDatabaseConnector{}
		conn, err := connector.Connect()
		if err != nil {
			t.Fatal(err)
		}
		strg, err := storage.NewMarkClient(conn)
		if err != nil {
			t.Fatal(err)
		}

		out, err := testutils.CmdExecute(NewRootCmd(strg), args...)
		if err != nil {
			t.Fatal(err)
		}

		cmdAsString := "aerospace-marks " + strings.Join(args, " ") + "\n"
		snaps.MatchSnapshot(t, cmdAsString, out)
	})
}
