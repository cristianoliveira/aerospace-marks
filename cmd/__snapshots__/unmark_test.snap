
[TestUnmarkCommand/unmarks_a_mark_from_a_window_-_`marks_unmark_mark1` - 1]
aerospace-marks unmark mark1

Removed 1 marks

---

[TestUnmarkCommand/unmarks_all_marks_from_a_window_-_`marks_unmark` - 1]
aerospace-marks unmark

Removed 2 marks

---

[TestUnmarkCommand/unmarks_--help - 1]
aerospace-marks unmark --help

Unmark one or more windows by identifier.

unmark [<identifier>]

unmark cmd will remove identifier from the list of current marks on a window. If identifier is omitted, all marks are removed.

Usage:
  aerospace-marks unmark [flags]

Flags:
  -h, --help   help for unmark

---
